configs/maskformer/maskformer_r50_512x512_80k_ade20k.py

```diff
@@ -0,0 +1,135 @@
+            num_layers=6,
+            transformerlayers=dict(
+                type='DetrTransformerDecoderLayer',
+                attn_cfgs=dict(
+                    type='MultiheadAttention',
+                    embed_dims=256,
+                    num_heads=8,
+                    attn_drop=0.1,
+                    proj_drop=0.1,
+                    dropout_layer=None,
+                    batch_first=False),
+                ffn_cfgs=dict(
+                    embed_dims=256,
+                    feedforward_channels=2048,
+                    num_fcs=2,
+                    act_cfg=dict(type='ReLU', inplace=True),
+                    ffn_drop=0.1,
+                    dropout_layer=None,
+                    add_identity=True),
+                # the following parameter was not used,
+                # just make current api happy
+                feedforward_channels=2048,
+                operation_order=('self_attn', 'norm', 'cross_attn', 'norm',
+                                 'ffn', 'norm')),
+            init_cfg=None),
+        loss_cls=dict(
+            type='CrossEntropyLoss',
+            use_sigmoid=False,
+            loss_weight=1.0,
+            reduction='mean',
+            class_weight=[1.0] * num_classes + [0.1]),
+        loss_mask=dict(
+            type='FocalLoss',
+            use_sigmoid=True,
+            gamma=2.0,
+            alpha=0.25,
+            reduction='mean',
+            loss_weight=20.0),
+        loss_dice=dict(
+            type='DiceLoss',
+            use_sigmoid=True,
+            activate=True,
+            reduction='mean',
+            naive_dice=True,
+            eps=1.0,
+            loss_weight=1.0),
+        train_cfg=dict(
+            assigner=dict(
+                type='HungarianAssigner',
+                match_costs=[
+                    dict(type='ClassificationCost', weight=1.0),
+                    dict(type='FocalLossCost', weight=20.0, binary_input=True),
+                    dict(type='DiceCost', weight=1.0, pred_act=True, eps=1.0)
+                ]),
+            sampler=dict(type='MaskPseudoSampler')),
+        test_cfg=dict(
+            panoptic_on=False,
+            # For now, the dataset does not support
+            # evaluating semantic segmentation metric.
+            semantic_on=True,
+            instance_on=False,
+            # max_per_image is for instance segmentation.
+            max_per_image=100,
+            object_mask_thr=0.8,
+            iou_thr=0.8,
+            # In MaskFormer's panoptic postprocessing,
+            # it will not filter masks whose score is smaller than 0.5 .
+            filter_low_score=False)),
+    # training and testing settings
+    train_cfg=dict(),
+    test_cfg=dict(mode='whole'),
+)
+
+# optimizer
+optimizer = dict(
+    type='AdamW', lr=0.0001, betas=(0.9, 0.999), weight_decay=0.0005)
+optim_wrapper = dict(
+    _delete_=True,
+    type='OptimWrapper',
+    optimizer=optimizer,
+    clip_grad=None,
```

Comment:
clip grad need to be checked

https://github.com/facebookresearch/MaskFormer/blob/da3e60d85fdeedcb31476b5edd7d328826ce56cc/configs/ade20k-150/Base-ADE20K-150.yaml#L29-L33